src/argilla/client/feedback/dataset/base.py

```diff
@@ -349,36 +356,35 @@ def _validate_records(self, records: List[FeedbackRecord]) -> None:
         Raises:
             ValueError: if the `fields` schema does not match the `FeedbackRecord.fields` schema.
         """
-        if self._fields_schema is None:
-            self._fields_schema = generate_pydantic_schema_for_fields(self.fields)
-
-        # TODO: this is here to avoid conflicts with other PRs
-        if not hasattr(self, "_metadata_schema"):
-            self._metadata_schema = None
-
-        # TODO: review up to what extent we want to apply the validation here
-        existing_metadata_properties = self.metadata_properties
-        if self._metadata_schema is None and existing_metadata_properties:
-            self._metadata_schema = generate_pydantic_schema_for_metadata(
-                existing_metadata_properties, allow_extra_metadata=self._allow_extra_metadata
-            )
+        # WE build once the schemas to avoid building it for each record
+        fields_schema = self._build_fields_schema()
+        metadata_schema = self._build_metadata_schema()
```

Comment:
This should just be built if there are any `metadata_properties`, otherwise there's no need to

Reply:
We need to check the  `allow_extra_metadata` which is configured as part of the `MetaConfig` inside the metadata schema. If we generate the schema only when there are metadata properties, we need to add extra steps checking the allow_metadata_extra in the validation which makes the code very dirty